server/controllers/SolutionController.js

```diff
@@ -128,10 +128,32 @@ exports.create = async (req, res) => {
     diagram,
   });
 
+  const challenge = await Challenge.findByPk(challengeId);
+
+  // Update user score
+  try {
+    const user = await User.findByPk(userId);
+    if (!user) {
+      return res.status(404).json({ message: "User not found" });
+    }
+
+    if (challenge.difficulty === "easy") {
+      user.score += 10;
+    } else if (challenge.difficulty === "medium") {
+      user.score += 20;
+    } else {
```

Comment:
This can cause problems if we add more difficulties in the future, or the difficulty is invalid. Check for hard and on `else` print an error.